cmd/operator-sdk/alpha/bundle/build.go

```diff
@@ -0,0 +1,90 @@
+		Long: `The 'operator-sdk bundle build' command will build an operator
+bundle image containing operator metadata and manifests, tagged with the
+provided image tag. Metadata generated by this command defines this image
+and should not be modified.
+
+If you wish to use a non-default image builder, run the 'generate' subcommand
+and build the operator's bundle image manually.
+
+NOTE: bundle images are not runnable.
+`,
+		Example: `The following command will build a test-operator bundle image using Docker.
+This image will contain manifests for package channels 'stable' and 'beta':
+
+$ operator-sdk bundle build \
+    --directory ./deploy/olm-catalog/test-operator \
+    --tag quay.io/example/operator:v0.1.0 \
+    --package test-operator \
+    --channels stable,beta \
+    --default stable \
+    --overwrite
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Clean up transient metadata and Dockerfile once the image is built,
+			// as they are no longer needed.
+			metaDir := filepath.Join(c.directory, "metadata")
+			_, err := os.Stat(metaDir)
+			metaExists := os.IsExist(err)
+			dockerFile := filepath.Join(c.directory, "Dockerfile")
+			_, err = os.Stat(dockerFile)
+			dockerFileExists := os.IsExist(err)
+			defer func() {
+				if !metaExists {
+					_ = os.RemoveAll(metaDir)
+				}
+				if !dockerFileExists {
+					_ = os.RemoveAll(dockerFile)
+				}
+			}()
+			return bundle.BuildFunc(c.directory, c.imageTag, c.imageBuilder,
+				c.packageName, c.channels, c.channelDefault, c.overwrite)
+		},
+	}
+
+	cmd.Flags().StringVarP(&c.directory, "directory", "d", "",
+		"The directory where bundle manifests are located")
+	if err := cmd.MarkFlagRequired("directory"); err != nil {
+		log.Fatalf("Failed to mark `directory` flag for `build` subcommand as required")
+	}
+	cmd.Flags().StringVarP(&c.packageName, "package", "p", "",
+		"The name of the package that bundle image belongs to")
+	if err := cmd.MarkFlagRequired("package"); err != nil {
+		log.Fatalf("Failed to mark `package` flag for `build` subcommand as required")
+	}
+	cmd.Flags().StringVarP(&c.channels, "channels", "c", "",
+		"The list of channels that bundle image belongs to")
+	if err := cmd.MarkFlagRequired("channels"); err != nil {
+		log.Fatalf("Failed to mark `channels` flag for `build` subcommand as required")
+	}
+	cmd.Flags().StringVarP(&c.imageTag, "tag", "t", "",
+		"The image tag applied to the bundle image")
+	if err := cmd.MarkFlagRequired("tag"); err != nil {
+		log.Fatalf("Failed to mark `tag` flag for `build` subcommand as required")
+	}
+
+	cmd.Flags().StringVarP(&c.imageBuilder, "image-builder", "b", "docker",
+		"Tool to build container images. One of: [docker, podman, buildah]")
+	cmd.Flags().StringVarP(&c.channelDefault, "default", "e", "",
+		"The default channel for the bundle image")
+	cmd.Flags().BoolVarP(&c.overwrite, "overwrite", "o", false,
+		"To overwrite annotations.yaml locally if existed. By default, overwrite is set to `false`.")
```

Comment:
If `annotations.yaml` already exists, will `bundle build` use it? Or will `bundle build` always generate `annotations.yaml` from scratch?

Also, what if `annotations.yaml` is present and `--overwrite=false`?

Reply:
[`bundle.BuildFunc`](https://github.com/operator-framework/operator-registry/blob/918955213adf5b2525deaa167f0c4a4c05f491c6/pkg/lib/bundle/build.go#L43-L46) will create an `annotations.yaml` if not present, and use a local file if present.

Reply:
Just making sure that's the desired behavior. I know there was discussion about making the `build` subcommand entirely hermetic and never using existing file.

Reply:
@dmesser ^

Reply:
Annotations.yaml contain the metadata for channels, this is not something that the user will set and want to re-use. They may change it between builds, or on different branches but we should NOT be creating a new one every time. 

What are the defaults for the annotations.yaml?

Reply:
@shawn-hurley I thought we did not intend to persist metadata on disk for a vanilla `build` invocation, which is why we have `build --generate-only` in case `annotations.yaml` or the Dockerfile need to be modified; in the latter case we do not guarantee anything about the resulting bundle image since it was manually modified and build, i.e. "use --generate-only at your own risk".

Reply:
Resolution: if metadata or a Dockerfile exists on disk, use these; otherwise use transient metadata/Dockerfile (do not write to disk).